templates/sync-function/comparison-module.js

```diff
@@ -0,0 +1,292 @@
+function comparisonModule(timeModule, buildItemPath) {
+  return {
+    validateMinValueInclusiveConstraint: validateMinValueInclusiveConstraint,
+    validateMaxValueInclusiveConstraint: validateMaxValueInclusiveConstraint,
+    validateMinValueExclusiveConstraint: validateMinValueExclusiveConstraint,
+    validateMaxValueExclusiveConstraint: validateMaxValueExclusiveConstraint,
+    validateImmutable: validateImmutable,
+    validateEquality: validateEquality
+  };
+
+  function validateMinValueInclusiveConstraint(itemStack, rangeConstraint, validatorType) {
```

Comment:
The following code in this file has been lifted from the validation module with only minor edits.